sympy/functions/combinatorial/numbers.py

```diff
@@ -1400,6 +1401,181 @@ def _eval_is_positive(self):
+
+    * ``motzkin(n)`` gives the `n^{th}` Motzkin number.
+
+    * ``find_first_n_motzkin(n)`` gives the first
+    `n` Motzkin numbers.
+
+    * ``find_motzkin_numbers_in_range(n)`` gives
+    the all Motzkin numbers in the interval `[x, y]`.
+
+    * ``is_motzkin(n)`` returns the position of
+    `n` in the Motzkin sequence of numbers if `n`
+    is a Motzkin number, else returns False
+
+    * ``is_prime(n)`` returns True if `n` is
+    a Motzkin prime number, else returns False
+
+    Examples
+    ========
+
+    >>> from sympy import motzkin
+    >>> motzkin(23)
+    1129760415
+    >>> motzkin(12.132)
+    17673.6827206284 + 0.00229270467798908*I
+
+    >>> motzkin.find_first_n_motzkin(15)
+    [1, 1, 2, 4, 9, 21, 51, 127, 323, 835, 2188, 5798, 15511, 41835, 113634]
+    >>> motzkin.find_motzkin_numbers_in_range(50, 100000)
+    [51, 127, 323, 835, 2188, 5798, 15511, 41835]
+    >>> motzkin.is_motzkin(3192727797)
+    24
+    >>> motzkin.is_motzkin(486431896)
+    False
+    >>> motzkin.is_prime(953467954114363)
+    True
+
+    >>> from sympy import Symbol, Sum, binomial, hyper
+    >>> n = Symbol('n')
+
+    >>> motzkin(n).rewrite(Sum)
+    Sum(binomial(n, 2*_k)*catalan(_k), (_k, 0, floor(n/2)))
+
+    >>> motzkin(n).rewrite(binomial)
+    Sum(binomial(2*_k, _k)*binomial(n, 2*_k)/(_k + 1), (_k, 0, floor(n/2)))
+
+    >>> motzkin(n).rewrite(hyper)
+    hyper((0.5 - 0.5*n, -0.5*n), (2,), 4)
+
+    See Also
+    ========
+
+    bell, bernoulli, binomial, catalan, euler, fibonacci, tribonacci, harmonic, lucas,
+    genocchi, partition, sympy.functions.special.hyper.hyper
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Motzkin_number
+    .. [2] https://mathworld.wolfram.com/MotzkinNumber.html
+    .. [3] https://oeis.org/A001006
+
+    """
+
+    @classmethod
+    def eval(cls, n):
+        n = sympify(n)
+        from sympy.functions.special.hyper import hyper
+        if not n.is_symbol:
+            m = hyper((0.5*(1 - n), -0.5*n), [2], 4).evalf()
+            if n.is_Integer and n.is_nonnegative:
+                return int(m)
+            else:
+                return m
+
+    @staticmethod
+    def find_first_n_motzkin(n):
+        n = sympify(n)
+        motzkins = []
+        if n.is_symbol:
+            raise TypeError('Expected number, not %s'%(type(n)))
+        for i in range(int(n)):
+            motzkins.append(motzkin(i))
+        return motzkins
+
+    @staticmethod
+    def is_motzkin(n):
+        n = sympify(n)
+        if n.is_symbol:
+            raise TypeError('Expected number, not %s'%(type(n)))
+        if n.is_Integer and n.is_positive:
+            i = 0
+            num = motzkin(i)
+            while num <= n:
+                if num == n:
+                    return i
+                i += 1
+                num = motzkin(i)
+            return False
+        else:
+            return False
+
+    @staticmethod
+    def find_motzkin_numbers_in_range(x, y):
```

Comment:
I'm also not sure if this needs to be here. motzkin is nondecreasing, so this isn't hard to compute.

Reply:
I thought it would be useful. I'll remove it if you feel its not required